cip-0005.md

```diff
@@ -0,0 +1,82 @@
+<pre>
+  CIP: 5
+  Title: Voting Meta Protocol through Broadcasts
+  Authors: Ruben de Vries
+  Discussions-To: https://counterpartytalk.org/t/cip-proposal-voting-meta-protocol-through-broadcasts/1926
+  Status: Draft
+  Type: Standards
+  Created: 2016-03-21
+</pre>
+
+## Abstract ##
+
+Any user should be able to initiate a vote for all holders of a Counterparty asset.
+Holders of the asset can broadcast their vote and all Counterparty nodes can tally the votes.
+
+## Motivation ##
+
+With the lack of PoW 'voting' by miners there's no good way to reach / poll for consensus on changes to the protocol
+that aren't completely uncontroversial.
+Using this proposed protocol, Counterparty developers can initiate a vote to all XCP holders to learn the will of the community.
+
+## Rationale ##
+
+Being able to let all XCP owners vote on a particular issue can be very useful to poll the consensus / desire of the community.
+
+The votes done in this way is not part of this CIP and the initial intention is not to have any protocol changes activate automatically based on these votes,
+though that could be a future CIP if there's a desire to do so.
+
+As discussed in the community, to avoid making anyone have more power within the protocol, we won't be using tokens
+but will instead use a meta protocol with messages encoded in broadcast messages.
+
+This also makes this easily usable for other assets.
+
+We'll add the code to track votes to counterparty-lib/client so that anyone can easily track ongoing votes without requiring extra scripts.
+We'll also add code to counterwallet to easily cast votes, this should cover a big part of the XCP holders.
+
+## Specification ##
+
+There will be 2 types of 'special' broadcast messages that are used for this; `INITVOTE` and `CASTVOTE`.
+
+#### `INITVOTE`
+```INITVOTE <VOTENAME> <ASSET> <DEADLINE> (<STAKEHEIGHT>)? OPTS <OPTION_1> <OPTION_N>```
+ - `VOTENAME`: any sequence of non-space characters used as label for the vote, must be unique across the network.
+ - `ASSET`: the asset for which the holders can vote.
+ - `DEADLINE`: the deadline (block height or timestamp of block, see below) for the voting period (int).
+ - `STAKEHEIGHT`: the block height at which to take the balances of `ASSET` to use to weight the votes.
+   if `STAKEHEIGHT` is ommitted then the block in which the `INITVOTE` broadcast is confirmed is used.
+ - `OPTS`, followed by 2+ options, ie; `OPTS TRUE FALSE`.
+   options are seperated by a space, any other characters are allowed but for simplicity only using alphanumeric chars is recommended.
```

Comment:
I think it's good to limit it for sanity/consistency at the UI level if nothing else, so that you can design the UIs to handle and look good with options up to that max length.